### PR TITLE
ct-fetch: fix nil pointer dereference

### PR DESCRIPTION
**Pull request by jschanck**

### CODE REVIEW

File: go/cmd/ct-fetch/ct-fetch.go [hunk 1]
```diff
@@ -504,7 +504,9 @@ func (ld *LogSyncEngine) NewLogWorker(ctx context.Context, ctLogMeta *types.CTLo
 	}
 
 	metricKey := ctLogMeta.MetricKey()
-	metrics.SetGauge([]string{metricKey, "coverage"}, float32(logObj.MaxEntry-logObj.MinEntry+1)/float32(sth.TreeSize))
+	if sth != nil {
```
Comment (mozkeeler) [inline go/cmd/ct-fetch/ct-fetch.go:507]: Don't we already dereference `sth` on e.g. line 478?

Comment (mozkeeler) [reply go/cmd/ct-fetch/ct-fetch.go:507]: Ohh, right, not if `fetchErr` is something...

Comment (jschanck) [reply go/cmd/ct-fetch/ct-fetch.go:507]: Right, `sth` is only `nil` if `fetchErr` is not.

**Review by mozkeeler — APPROVED** [review]
LGTM. It seems like in other places we use `fetchErr == nil` as the "is-sth-valid" guard, but I'm not sure that's better.

**Pull request merged**